# Whom?

[![GitHub License](https://img.shields.io/github/license/Dragon1573/PyChallenge-Tips?color=important&label=Licence&style=flat-square)](https://github.com/Dragon1573/PyChallenge-Tips/blob/master/LICENSE)
[![Jump to Level 15](https://img.shields.io/badge/Jump%20to-Level%2015-blue?style=flat-square)](http://www.pythonchallenge.com/pc/return/uzi.html)

<center><img src="../../resources/imgs/Quiz15-1.png" /></center>

&emsp;&emsp;通过关卡图片，我们可以获得以下重要信息：

- 当年是1xx6年
- 当年1月26日为周一
- 放大图片右下角，二月有29天，因此当年是闰年

&emsp;&emsp;获取关卡源代码，进一步分析线索。

In [1]:
from requests import post
from bs4 import BeautifulSoup as Soup

In [2]:
response = post(
    'http://www.pythonchallenge.com/pc/return/uzi.html',
    headers={'Authorization': 'Basic aHVnZTpmaWxl'}
)
response = Soup(response.text, features='html.parser')
print(response.prettify())

<html>
 <head>
  <title>
   whom?
  </title>
  <link href="../style.css" rel="stylesheet" type="text/css"/>
 </head>
 <body>
  <br/>
  <center>
   <!-- he ain't the youngest, he is the second -->
   <img src="screen15.jpg"/>
   <br/>
  </center>
 </body>
</html>
<!-- todo: buy flowers for tomorrow -->



&emsp;&emsp;`<center />`标签中有两句注释：

1. 他不是年龄最小的，他是第2小的
2. 待办事项：为明天买花

&emsp;&emsp;它们也是关键提示。因为符合以上3个条件的年份可能不止1个，还需要通过这个提示进一步筛选。

In [3]:
import calendar
import datetime

In [4]:
years = []
for year in range(1006, 1997, 10):
    # 获取日期
    date = datetime.datetime(year, 1, 27)
    # 闰年和星期判断
    if calendar.isleap(year) and date.weekday() == 1:
        years.append(year)
# 逆序排列
years.sort(reverse=True)
# 获得第2大（年龄第2小）的值
print(years[1])

1756


&emsp;&emsp;公元1756年1月27日，世界著名音乐大师沃尔夫冈·阿玛多伊斯·**莫扎特**出生，因此下一关的链接为<http://www.pythonchallenge.com/pc/return/mozart.html>。